dbt/include/athena/macros/materializations/models/incremental/merge.sql

```diff
@@ -29,7 +82,11 @@
     when matched
       then update set
         {% for col in update_columns %}
-          {{ '"' + col + '"' }} = {{ 'src."' + col + '"' }} {{ "," if not loop.last }}
+          {% if merge_update_columns_rules and col.name in merge_update_columns_rules %}
+            {{ get_update_statement(col, merge_update_columns_rules[col.name], loop.last) }}
+          {% else %}
+            {{ get_update_statement(col, merge_update_columns_default_rule, loop.last) }}
```

Comment:
in order to have this pr merged, we need to garantee that the behavior `{{ '"' + col + '"' }} = {{ 'src."' + col + '"' }} {{ "," if not loop.last }}` is still kept it. This means that in your `get_update_statement ` the should be a final else statement that columns are pick from the target table.

Reply:
if `merge_update_columns_default_rule == 'replace'` (which is the default) it will return
```
{{ col.quoted }} = {{ 'src.' + col.quoted }} {{ "," if not is_last }}
```

Reply:
```
{{ config(
    materialized='incremental',
    format='iceberg',
    incremental_strategy='merge',
    unique_key = ['id'],
    partitioned_by=['partition_timestamp'],
    merge_exclude_columns=['first_timestamp'],
) }}

SELECT
    '0' AS "id",
    0.3 AS "score",
    ARRAY [1] AS "array_field",
    'std_behaviour' AS"std_behaviour",
    'do_not_change_me' AS "keep_me_unchanged",
    2 AS "custom_rule",
    current_date AS "first_timestamp",
    current_date AS "partition_timestamp"
```

this model returns this query:

```
    merge into profiles.merge_test as target using profiles.merge_test__dbt_tmp as src
    ON (
      
        target.id = src.id
        
      
    )
    when matched
      then update set
        
          
            
    
        "score" = src."score" ,
    

          
        
          
            
    
        "array_field" = src."array_field" ,
    

          
        
          
            
    
        "std_behaviour" = src."std_behaviour" ,
    

          
        
          
            
    
        "keep_me_unchanged" = src."keep_me_unchanged" ,
    

          
        
          
            
    
        "custom_rule" = src."custom_rule" ,
    

          
        
          
            
    
        "partition_timestamp" = src."partition_timestamp" 
    

          
        
    when not matched
      then insert ("id", "score", "array_field", "std_behaviour", "keep_me_unchanged", "custom_rule", "first_timestamp", "partition_timestamp")
       values (src."id", src."score", src."array_field", src."std_behaviour", src."keep_me_unchanged", src."custom_rule", src."first_timestamp", src."partition_timestamp");
```

Reply:
this model
```
{{ config(
    materialized='incremental',
    format='iceberg',
    incremental_strategy='merge',
    unique_key = ['id'],
    partitioned_by=['partition_timestamp'],
    merge_exclude_columns=['first_timestamp'],
    merge_update_columns_rules={
        'score': 'sum',
        'array_field': 'append',
        'keep_me_unchanged': 'coalesce',
        'custom_rule': '_old_ + _old_ + _new_'
    },
) }}

SELECT
    '0' AS "id",
    0.3 AS "score",
    ARRAY [1] AS "array_field",
    'std_behaviour' AS"std_behaviour",
    'do_not_change_me' AS "keep_me_unchanged",
    2 AS "custom_rule",
    current_date AS "first_timestamp",
    current_date AS "partition_timestamp"
```

returns this query:
```
    merge into profiles.merge_test as target using profiles.merge_test__dbt_tmp as src
    ON (
      
        target.id = src.id
        
      
    )
    when matched
      then update set
        
          
            
    
        "score" = src."score" + target."score" ,
      

          
        
          
            
    "array_field" = src."array_field" || target."array_field" ,

          
        
          
            
    
        "std_behaviour" = src."std_behaviour" ,
    

          
        
          
            
    
        "keep_me_unchanged" = coalesce(src."keep_me_unchanged", target."keep_me_unchanged") ,

          
        
          
            
    
        "custom_rule" = src."custom_rule" + src."custom_rule" + target."custom_rule" ,
    

          
        
          
            
    
        "partition_timestamp" = src."partition_timestamp" 
    

          
        
    when not matched
      then insert ("id", "score", "array_field", "std_behaviour", "keep_me_unchanged", "custom_rule", "first_timestamp", "partition_timestamp")
       values (src."id", src."score", src."array_field", src."std_behaviour", src."keep_me_unchanged", src."custom_rule", src."first_timestamp", src."partition_timestamp");
```

Reply:
ah cool, I missed that part.I will try to block some time to test and review the pr properly in order to get it merge and release soon.